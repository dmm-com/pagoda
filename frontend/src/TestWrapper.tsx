import { createTheme, ThemeProvider } from "@mui/material/styles";
import { SnackbarProvider } from "notistack";
import React, { FC } from "react";
import { MemoryRouter } from "react-router";

export const TestWrapper: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        <MemoryRouter>{children}</MemoryRouter>
      </SnackbarProvider>
    </ThemeProvider>
  );
};

export const TestWrapperWithoutRoutes: FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const theme = createTheme();
  return (
    <ThemeProvider theme={theme}>
      <SnackbarProvider maxSnack={1} autoHideDuration={100}>
        {children}
      </SnackbarProvider>
    </ThemeProvider>
  );
};
