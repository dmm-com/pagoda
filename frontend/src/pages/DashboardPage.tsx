import {
  Box,
  Container,
  Grid,
  List,
  ListItem,
  ListItemText,
  Typography,
  TypographyTypeMap
} from "@mui/material";
import { OverridableComponent } from "@mui/material/OverridableComponent";
import { styled } from "@mui/material/styles";
import React, { FC, useState } from "react";
import { Link, useNavigate, useLocation } from "react-router";

import { useAsyncWithThrow } from "../hooks/useAsyncWithThrow";

import { AironeBreadcrumbs } from "components/common/AironeBreadcrumbs";
import { CategoryListHeader } from "components/category/CategoryListHeader";
import { Loading } from "components/common/Loading";
import { PaginationFooter } from "components/common/PaginationFooter";
import { usePage } from "hooks/usePage";
import { useSimpleSearch } from "hooks/useSimpleSearch";
import { aironeApiClient } from "repository/AironeApiClient";
import { entityEntriesPath, entryDetailsPath } from "routes/Routes";
import { EntityListParam } from "services/Constants";

const StyledContainer = styled(Container)({
  marginTop: "16px",
});

const ResultBox = styled(Box)({
  marginTop: "40px",
  display: "flex",
  flexWrap: "wrap",
  gap: "32px",
});

const Result = styled(Typography)(({ theme }) => ({
  color: theme.palette.primary.main,
  textDecoration: "none",
  flexGrow: 1,
  maxWidth: theme.breakpoints.values.lg,
  overflow: "hidden",
  textOverflow: "ellipsis",
})) as OverridableComponent<TypographyTypeMap>;

const ResultEntityForEntry = styled(Typography)(({ }) => ({
  color: "gray",
}));

export const DashboardPage: FC = () => {
  const navigate = useNavigate();
  const location = useLocation();

  // variable to store search query for Category
  const params = new URLSearchParams(location.search);
  const [queryCategory, setQueryCategory] = useState<string>(params.get("query") ?? "");
  const [page, changePage] = usePage();

  // variable to store search query for Searching Entry
  const [query, submitQuery] = useSimpleSearch();

  const entries = useAsyncWithThrow(async () => {
    if (query != null) {
      return await aironeApiClient.getSearchEntries(query);
    }
  }, [location, query]);

  const entities = useAsyncWithThrow(async () => {
    return await aironeApiClient.getEntities(undefined, undefined, true);
  });

  // If there is only one search result, move to entry details page.
  if (!entries.loading && entries.value?.length === 1) {
    navigate(
      entryDetailsPath(entries.value[0].schema?.id ?? 0, entries.value[0].id)
    );
  }

  const [toggle, setToggle] = useState(false);

  // request handler when user specify query
  const handleChangeQuery = (newQuery?: string) => {
    changePage(1);
    setQueryCategory(newQuery ?? "");

    navigate({
      pathname: location.pathname,
      search: newQuery ? `?query=${newQuery}` : "",
    });
  };

  const categories = useAsyncWithThrow(async () => {
    return await aironeApiClient.getCategories(page, queryCategory);
  }, [page, query, toggle]);

  return (
    <Box>
      <AironeBreadcrumbs>
        <Typography color="textPrimary">Top</Typography>
      </AironeBreadcrumbs>

      <StyledContainer>

        {/* Context of Category */}
        <>
          <Grid container spacing={3}>
            {categories.value?.results.map((category) => (
              <Grid item md={4} key={category.id}>
                <List
                  subheader={
                    <CategoryListHeader
                      category={category}
                      setToggle={() => setToggle(!toggle)}
                    />
                  }
                >
                  <Box sx={{ overflowY: "scroll", maxHeight: 300 }}>
                    {category.models.map((models) => (
                      <ListItem
                        button
                        key={models.id}
                        component={Link}
                        to={entityEntriesPath(models.id)}
                      >
                        <ListItemText primary={models.name} />
                      </ListItem>
                    ))}
                  </Box>
                </List>
              </Grid>
            ))}
          </Grid>
          <PaginationFooter
            count={categories.value?.count ?? 0}
            maxRowCount={EntityListParam.MAX_ROW_COUNT}
            page={page}
            changePage={changePage}
          />
        </>

        {/* Models that are configured to be shown at the top page */}
        {entries.loading ? (
          <Loading />
        ) : entries.value ? (
          <ResultBox id="entry_list">
            {entries.value.map((entry) => (
              <Box id="entry" key={entry.id}>
                <Result
                  component={Link}
                  to={entryDetailsPath(entry.schema?.id ?? 0, entry.id)}
                >
                  {entry.name}
                </Result>
                <ResultEntityForEntry>
                  {entry.schema?.name}
                </ResultEntityForEntry>
              </Box>
            ))}
          </ResultBox>
        ) : entities.loading ? (
          <Loading />
        ) : (
          <ResultBox id="entity_list">
            {entities.value?.results?.map((entity) => (
              <Result
                key={entity.id}
                component={Link}
                to={entityEntriesPath(entity.id)}
              >
                {entity.name}
              </Result>
            ))}
          </ResultBox>
        )}
      </StyledContainer>
    </Box>
  );
};
