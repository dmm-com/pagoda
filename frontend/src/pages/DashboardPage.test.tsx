/**
 * @jest-environment jsdom
 */

import { act, render, screen, waitFor } from "@testing-library/react";
import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";

import { DashboardPage } from "./DashboardPage";

import { TestWrapper } from "TestWrapper";

const server = setupServer(
  // getEntities
  http.get("http://localhost/category/api/v2/", () => {
    return HttpResponse.json({
      count: 3,
      next: null,
      previous: null,
      results: [
        {
          id: 1,
          name: "aaa",
          note: "",
          models: [],
          priority: 20,
        },
        {
          id: 2,
          name: "aaaaa",
          note: "",
          models: [],
          priority: 10,
        },
        {
          id: 3,
          name: "bbbbb",
          note: "",
          models: [],
          priority: 0,
        },
      ],
    });
  }),
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("should match snapshot", async () => {
  Object.defineProperty(window, "django_context", {
    value: {
      user: {
        is_superuser: false,
      },
    },
    writable: false,
  });

  const result = await act(() => {
    return render(<DashboardPage />, {
      wrapper: TestWrapper,
    });
  });
  await waitFor(() => {
    expect(screen.queryByTestId("loading")).not.toBeInTheDocument();
  });
  expect(result).toMatchSnapshot();
});
