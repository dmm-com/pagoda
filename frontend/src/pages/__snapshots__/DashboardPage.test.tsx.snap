// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1br4z8z-MuiContainer-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-1abudc7"
          >
            <div
              class="MuiBox-root css-1m3p9o7"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-miwl14-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-4ama2z-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    placeholder="カテゴリを絞り込む"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
            >
              <ul
                class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
              >
                <div
                  class="MuiBox-root css-mbiude"
                >
                  <div
                    class="MuiBox-root css-1atl7vu"
                  >
                    <img
                      class="MuiBox-root css-ku8je2"
                      src="/static/images/category/01.png"
                    />
                    <div
                      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    >
                      aaa
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-1kcxdgd"
                />
              </ul>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
            >
              <ul
                class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
              >
                <div
                  class="MuiBox-root css-mbiude"
                >
                  <div
                    class="MuiBox-root css-1atl7vu"
                  >
                    <img
                      class="MuiBox-root css-ku8je2"
                      src="/static/images/category/01.png"
                    />
                    <div
                      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    >
                      aaaaa
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-1kcxdgd"
                />
              </ul>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
            >
              <ul
                class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
              >
                <div
                  class="MuiBox-root css-mbiude"
                >
                  <div
                    class="MuiBox-root css-1atl7vu"
                  >
                    <img
                      class="MuiBox-root css-ku8je2"
                      src="/static/images/category/01.png"
                    />
                    <div
                      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    >
                      bbbbb
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-1kcxdgd"
                />
              </ul>
            </div>
          </div>
          <div
            class="MuiBox-root css-d3lzg9"
            id="pagination"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              1 - 3 / 3 件
            </p>
            <div
              class="MuiStack-root css-1sazv7p-MuiStack-root"
            >
              <nav
                aria-label="pagination navigation"
                class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
              >
                <ul
                  class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateBeforeIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-current="true"
                      aria-label="page 1"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                      tabindex="0"
                      type="button"
                    >
                      1
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateNextIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1br4z8z-MuiContainer-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-1abudc7"
        >
          <div
            class="MuiBox-root css-1m3p9o7"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-miwl14-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-4ama2z-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SearchIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </div>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  placeholder="カテゴリを絞り込む"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
          >
            <ul
              class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
            >
              <div
                class="MuiBox-root css-mbiude"
              >
                <div
                  class="MuiBox-root css-1atl7vu"
                >
                  <img
                    class="MuiBox-root css-ku8je2"
                    src="/static/images/category/01.png"
                  />
                  <div
                    class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    aaa
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1kcxdgd"
              />
            </ul>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
          >
            <ul
              class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
            >
              <div
                class="MuiBox-root css-mbiude"
              >
                <div
                  class="MuiBox-root css-1atl7vu"
                >
                  <img
                    class="MuiBox-root css-ku8je2"
                    src="/static/images/category/01.png"
                  />
                  <div
                    class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    aaaaa
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1kcxdgd"
              />
            </ul>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
          >
            <ul
              class="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
            >
              <div
                class="MuiBox-root css-mbiude"
              >
                <div
                  class="MuiBox-root css-1atl7vu"
                >
                  <img
                    class="MuiBox-root css-ku8je2"
                    src="/static/images/category/01.png"
                  />
                  <div
                    class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    bbbbb
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-1kcxdgd"
              />
            </ul>
          </div>
        </div>
        <div
          class="MuiBox-root css-d3lzg9"
          id="pagination"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            1 - 3 / 3 件
          </p>
          <div
            class="MuiStack-root css-1sazv7p-MuiStack-root"
          >
            <nav
              aria-label="pagination navigation"
              class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
            >
              <ul
                class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
              >
                <li>
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateBeforeIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                  </button>
                </li>
                <li>
                  <button
                    aria-current="true"
                    aria-label="page 1"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                    tabindex="0"
                    type="button"
                  >
                    1
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateNextIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
